packages/dds/tree/src/feature-libraries/editable-tree/editableTree.ts

```diff
@@ -0,0 +1,434 @@
+ *
+ * TODO:
+ * EditableTree should provide easy access to children in a way thats guaranteed
+ * not to do this unwrapping for cases which need to refer to the actual nodes.
+ * This may include cases like creating anchors and/or editing.
+ */
+export type UnwrappedEditableTree = EditableTreeOrPrimitive | readonly UnwrappedEditableTree[];
+
+/**
+ * A field of an {@link EditableTree}.
+ */
+export type EditableField = readonly [FieldSchema, readonly EditableTree[]];
+
+/**
+ * Unwraps fields with non-sequence multiplicities.
+ */
+export type UnwrappedEditableField = UnwrappedEditableTree | undefined | readonly UnwrappedEditableTree[];
+
+export interface EditableTreeContext {
+    /**
+     * Call before editing.
+     *
+     * Note that after performing edits, EditableTrees for nodes that no longer exist are invalid to use.
+     * TODO: maybe add an API to check if a specific EditableTree still exists,
+     * and only make use other than that invalid.
+     */
+    prepareForEdit(): void;
+
+    /**
+     * Call to free resources.
+     * EditableTrees created in this context are invalid to use after this.
+     */
+    free(): void;
+}
+
+class ProxyContext implements EditableTreeContext {
+    public readonly withCursors: Set<ProxyTarget> = new Set();
+    public readonly withAnchors: Set<ProxyTarget> = new Set();
+    constructor(public readonly forest: IEditableForest) {}
+
+    public prepareForEdit(): void {
+        for (const target of this.withCursors) {
+            target.prepareForEdit();
+        }
+        assert(this.withCursors.size === 0, "prepareForEdit should remove all cursors");
+    }
+    public free(): void {
+        for (const target of this.withCursors) {
+            target.free();
+        }
+        for (const target of this.withAnchors) {
+            target.free();
+        }
+        assert(this.withCursors.size === 0, "free should remove all cursors");
+        assert(this.withAnchors.size === 0, "free should remove all anchors");
+    }
+}
+
+class ProxyTarget {
+	private readonly lazyCursor: ITreeSubscriptionCursor;
+    private anchor?: Anchor;
+
+	constructor(
+		public readonly context: ProxyContext,
+		cursor: ITreeSubscriptionCursor,
+	) {
+		this.lazyCursor = cursor.fork();
+        context.withCursors.add(this);
+	}
+
+    public free(): void {
+        this.lazyCursor.free();
+        this.context.withCursors.delete(this);
+        if (this.anchor !== undefined) {
+            this.context.forest.anchors.forget(this.anchor);
+            this.context.withAnchors.delete(this);
+            this.anchor = undefined;
+        }
+    }
+
+    public prepareForEdit(): void {
+        if (this.anchor === undefined) {
+            this.anchor = this.lazyCursor.buildAnchor();
+            this.context.withAnchors.add(this);
+        }
+        this.lazyCursor.clear();
+        this.context.withCursors.delete(this);
+    }
+
+	public get cursor(): ITreeSubscriptionCursor {
```

Comment:
mixed tabs and spaces. We use 4 space indent in this project currently. Please convert the indentation to spaces.

Reply:
probably, vs code issue. fixed.